# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/wasm/WasmTypes.h
# Commit: f981949ae991
# Full Hash: f981949ae991960c2099ea85435ddbb660a4229c
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-10-31 21:05:17
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 3: Implement the algorithm for obtaining tls and use it for wasm signal handling. r=lth
#   
#   This is the third part of series of patches to Frame without tls pointer.
#   Here we preserve initial tls in all entry stubs and then use it to find a proper tls instance for a given frame.
#   
# ==============================================================================

diff -r 6d85470024c3 -r f981949ae991 js/src/wasm/WasmTypes.h
--- a/js/src/wasm/WasmTypes.h	Sat Oct 31 09:56:05 2020 +0000
+++ b/js/src/wasm/WasmTypes.h	Sat Oct 31 09:56:05 2020 +0000
@@ -2585,6 +2585,7 @@
   }
   uint32_t lineOrBytecode() const { return lineOrBytecode_; }
   Kind kind() const { return Kind(kind_); }
+  bool mightBeCrossInstance() const { return kind() == CallSiteDesc::Dynamic; }
 };
 
 class CallSite : public CallSiteDesc {
@@ -3304,10 +3305,13 @@
 static_assert(!std::is_polymorphic_v<Frame>, "Frame doesn't need a vtable.");
 
 class FrameWithTls : public Frame {
- public:
   TlsData* calleeTls_;
   TlsData* callerTls_;
 
+ public:
+  TlsData* calleeTls() { return calleeTls_; }
+  TlsData* callerTls() { return callerTls_; }
+
   constexpr static uint32_t sizeWithoutFrame() {
     return sizeof(wasm::FrameWithTls) - sizeof(wasm::Frame);
   }
